### DISCUSSION TITLE
Which kind of extension should I implement?

### DISCUSSION THREAD

**Discussion by Pachat** [directus/directus#13569] [Archive]
For some reason, I'd show to the user the list of a junction table.
When clicking on one line, I'd like to achieve the following

1. Update a field (that I created) in the current user's row in `Directus_users`
2. Redirect the user to see, not the detail of the junction row, but the detail of a row in another collection, (which id is based on the junction row clicked)

What would be the appropriate extension to build?

Or should I wait "Add Data Flows to Directus"  #12522?

**Comment by rijkvanzanten** [comment]
I don't think there's currently a good extension type that allows this sort of functional override on an existing bit of UI. I think your best bet would be to create a custom interface that has that logic included.

**Reply by Pachat** [reply]
Thank you for your suggestion.
Such an interface should not introduce a different look and feel for the end user.
How can I mimic in the interface, let say for instance, the way Directus shows a list and its buttons, etc.

**Comment by Pachat** [comment]
I'm struggling with the Recipes endpoint example adapted so
```
export default (router, { services, exceptions }) => {
	const { ItemsService } = services;
	const { ServiceUnavailableException } = exceptions;

	router.get('/', (req, res, next) => {
		const networkService = new ItemsService('network', { schema: req.schema, accountability: req.accountability });

		networkService
			.readByQuery({ sort: [{ column: 'network', order: 'asc' }], fields: ['*'] })
			.then((results) => res.json(results))
			.catch((error) => {
				return next(new ServiceUnavailableException(error.message));
			});
	});
};

```
I could have the Hello example to work properly.
But with this code above, what is the url that gives results?
And how can I manage to read this result inside an interface extension and take some decisions?

I'd like this interface (read in the detail form) to update some value in the database on a click.
Is that possible?
